src/primary_names.lua

```diff
@@ -58,9 +58,13 @@ function primaryNames.createPrimaryNameRequest(name, initiator, timestamp, msgId
 	fundFrom = fundFrom or "balance"
 	local baseName = baseNameForName(name)
 
-	--- existing request for primary name from wallet?
-	local existingRequest = primaryNames.getPrimaryNameRequest(name)
-	assert(not existingRequest, "Primary name request for '" .. name .. "' already exists") -- TODO: should we error here or just let them create a new request and pay the fee again?
+	--- check the primary name request for the initiator does not already exist for the same name
+	--- this allows the caller to create a new request and pay the fee again, so long as it is for a different name
+	local existingRequest = primaryNames.getPrimaryNameRequest(initiator)
+	assert(
+		not existingRequest or existingRequest.name ~= name,
```

Comment:
Is any normalization needed for the names? Casing etc?

Reply:
we lower case all the names on the original request - can add same logic here jic

Reply:
enforced it again here - [2c49ac2](https://github.com/ar-io/ar-io-network-process/pull/230/commits/2c49ac2d168e7024a09f6c48d821447ccad212af)